sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala

```diff
@@ -261,6 +261,54 @@ class AstBuilder extends DataTypeAstBuilder
     WhileStatement(condition, body, Some(labelText))
   }
 
+  private def leaveOrIterateContextHasLabel(
+      ctx: RuleContext, label: String, isLeave: Boolean): Boolean = {
+    ctx match {
+      case c: BeginEndCompoundBlockContext
+        if isLeave &&
```

Comment:
shall we have a dedicated error for using ITERATE on the label of BEGIN END compound

Reply:
good idea, will do

Reply:
Done.